src/AppBundle/Controller/EventController.php

```diff
@@ -375,26 +387,32 @@ public function giveProxyAction(Event $event,Request $request,\Swift_Mailer $mai
             return $this->redirectToRoute('homepage');
         }
 
+        // proxy with a given beneficiary
         if ($request->get("beneficiary") > 0){
             $em = $this->getDoctrine()->getManager();
             $beneficiary = $em->getRepository('AppBundle:Beneficiary')->find($request->get("beneficiary"));
             if ($beneficiary){
-
                 $beneficiaries_ids = [];
                 foreach ($beneficiary->getMembership()->getBeneficiaries() as $b){
                     $beneficiaries_ids[] = $b;
                 }
                 /** @var Proxy $proxy */
-                $proxy = $em->getRepository('AppBundle:Proxy')->findOneBy(
+                $proxy = $em->getRepository('AppBundle:Proxy')->findBy(
                     array("owner" => $beneficiaries_ids, "event" => $event)
                 );
-
-                if ($proxy) {
-                    if ($proxy->getGiver() !== null) {
-                        $session->getFlashBag()->add('error', $beneficiary->getUser()->getFirstName() . ' accepte déjà de prendre une procuration d\'une autre personne');
+                $max_event_proxy_per_user = $this->container->getParameter("max_event_proxy_per_user");
+
+                if (count($proxy) == $max_event_proxy_per_user) {
+                    if ($max_event_proxy_per_user == 1) {
+                        if ($proxy->getGiver() !== null) {
```

Comment:
SI `proxy` est une collection, ca marche toujours ?

Reply:
j'ai réparé en rajoutant `->first()` :+1:

Reply:
edit : erreur "Call to a member function first() on array", j'ai remplacé par `[0]`